PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,880 @@
+        :returns: boolean
+        """
+        return (True if len(protocol) == 1 and (
+                'I' in protocol or 'F' in protocol) else False)
+
+    def get_object_components(self, regex_str, input_str):
+        """Get components from input string.
+
+        :param regex_str: the regex
+        :param input_str: the input string
+        :returns: dict
+        """
+        full_str = re.compile(regex_str)
+        match = full_str.match(input_str)
+        return match.groupdict() if match else None
+
+    def get_object_components_and_correct_host(self, regex_str, input_str):
+        """Get components from input string.
+
+        :param regex_str: the regex
+        :param input_str: the input string
+        :returns: dict
+        """
+        object_dict = self.get_object_components(regex_str, input_str)
+        if object_dict and 'host' in object_dict:
+            if object_dict['host'].endswith('-'):
+                object_dict['host'] = object_dict['host'][:-1]
+        return object_dict
+
+    def get_mv_component_dict(self, mv_name, revert=False):
+        """Get components from input string.
+
+        :param mv_name: the masking view name
+        :param revert: is it a revert back
+        :returns: dict
+        """
+        if revert:
+            regex_str = '^(?P<prefix>OS)-(?P<host>.+?)(?P<protocol>I|F)-' \
+                        '(?P<portgroup>(?!CD|RE|CD-RE).+)-(?P<postfix>MV)$'
+        else:
+            regex_str = '^(?P<prefix>OS)-(?P<host>.+?)((?P<srp>SRP.+?)-' \
+                        '(?P<slo>.+?)-(?P<workload>.+?)|(?P<no_slo>No_SLO))-' \
+                        '(?P<protocol>I|F)(?P<CD>-CD|s*)(?P<RE>-RE|s*)-' \
+                        '(?P<postfix>MV)$'
+        return self.get_object_components_and_correct_host(regex_str, mv_name)
+
+    def truncate_string(self, str_to_truncate, max_num):
+        """Truncate a string by taking first and last characters.
+
+        :param str_to_truncate: the string to be truncated
+        :param max_num: the maximum number of characters
+        :returns: string -- truncated string or original string
+        """
+        if len(str_to_truncate) > max_num:
+            new_num = len(str_to_truncate) - max_num // 2
+            first_chars = str_to_truncate[:max_num // 2]
+            last_chars = str_to_truncate[new_num:]
+            str_to_truncate = first_chars + last_chars
+        return str_to_truncate
+
+    def print_component_dict(self, mv_name, revert=False):
+        """Print the components to the screen.
+
+        :param mv_name: the masking view name
+        :param revert: is it a revert back
+        :returns: dict
+        """
+        if revert:
+            self.smart_print('\n', DEBUG)
+            self.smart_print(
+                'Checking if masking view is in the following format: ',
+                DEBUG)
+            self.smart_print(
+                '    OS-[shortHostName]-[protocol]-[portgroup_name]-MV',
```

Comment:
Would tabbing with `\t` help here instead of manual spacing? I see a few layers of indentation below, might be easier to manage consistency with varying amounts of `\t`